addons/mail/static/src/models/message/message.js

```diff
@@ -267,6 +287,33 @@ function factory(dependencies) {
             return `${this.modelName}_${data.id}`;
         }
 
+        /**
+         * @returns {boolean}
+         */
+        _computeCanBeDeleted() {
+            if (!session.is_admin && !this.isCurrentPartnerAuthor) {
+                return false;
+            }
+            const isMailChannelMessage = this.originThread && this.originThread.model === 'mail.channel'
+            if (isMailChannelMessage) {
+                return this.message_type === 'comment';
+            }
+            return this.is_note;
+        }
+
+        /**
+         * @returns {boolean}
+         */
+        _computeCanStarBeToggled() {
+            if (this.isTemporary) {
+                return false;
+            }
+            if (!this.isTransient) {
+                return true;
+            }
```

Comment:
this.isTemporary || this.isTransient
return false

Reply:
That is not equivalent to the previous condition in the template and will hide the star when the message is not transient  in most cases.

Reply:
I think this whole code was completely dumb then, first the condition on notification, and now this... I think that explains it, the "notification" condition was never reached because the message was always not transient in that case.

Reply:
So what, do I just `return !this.isTemporary && !this.isTransient` and remove the rest?

Reply:
I suppose you can try that indeed, if everything is green it should be fine, I don't see why that would be wrong. We just can't star messages that have no id in the server, but for all others it is possible (except functional reason and then it should be covered by a test).